src/Pact/Analyze/Analyze.hs

```diff
@@ -308,7 +314,7 @@ mkSymbolicCells tables = TableMap $ Map.fromList cellsList
              TyPrim TyTime    -> scTimeValues.at col    ?~ mkArray
              TyPrim TyString  -> scStringValues.at col  ?~ mkArray
              TyPrim TyKeySet  -> scKsValues.at col      ?~ mkArray
-             _ -> id -- error (show ty)
+             _                -> id -- error (show ty)
```

Comment:
should this be `error`?

Reply:
This case shouldn't be impossible, so we should use `Either` instead. This means that `mkSymbolicCells` is also `Either`, and then `mkInitialAnalyzeState` is too, and we discharge the `Left` in `checkFunctionBody`. I'll add a TODO for now

Reply:
Done